lib/tasks/openapi_generate.rake

```diff
@@ -0,0 +1,416 @@
+    {
+      "summary"     => "List #{klass_name.pluralize}#{" for #{primary_collection}" if primary_collection}",
+      "operationId" => "list#{primary_collection}#{klass_name.pluralize}",
+      "description" => "Returns an array of #{klass_name} objects",
+      "parameters"  => [
+        { "$ref" => "##{PARAMETERS_PATH}/QueryLimit"  },
+        { "$ref" => "##{PARAMETERS_PATH}/QueryOffset" }
+      ],
+      "responses"   => {
+        "200" => {
+          "description" => "#{klass_name.pluralize} collection",
+          "content"     => {
+            "application/json" => {
+              "schema" => { "$ref" => build_collection_schema(klass_name) }
+            }
+          }
+        }
+      }
+    }.tap do |h|
+      h["parameters"] << { "$ref" => build_parameter("ID") } if primary_collection
+    end
+  end
+
+  def build_collection_schema(klass_name)
+    collection_name = "#{klass_name.pluralize}Collection"
+    schemas[collection_name] = {
+      "type"       => "object",
+      "properties" => {
+        "meta"  => { "$ref" => "##{SCHEMAS_PATH}/CollectionMetadata" },
+        "links" => { "$ref" => "##{SCHEMAS_PATH}/CollectionLinks"    },
+        "data"  => {
+          "type"  => "array",
+          "items" => { "$ref" => build_schema(klass_name) }
+        }
+      }
+    }
+
+    "##{SCHEMAS_PATH}/#{collection_name}"
+  end
+
+  def openapi_schema(klass_name)
+    {
+      "type"       => "object",
+      "properties" => openapi_schema_properties(klass_name),
+    }
+  end
+
+  def openapi_schema_properties(klass_name)
+    model = klass_name.constantize
+    model.columns_hash.map do |key, value|
+      unless(GENERATOR_ALLOW_BLACKLISTED_ATTRIBUTES[key.to_sym] || []).include?(klass_name)
+        next if GENERATOR_BLACKLIST_ATTRIBUTES.include?(key.to_sym)
+      end
+
+      [key, openapi_schema_properties_value(klass_name, model, key, value)]
+    end.compact.sort.to_h
+  end
+
+  def openapi_schema_properties_value(klass_name, model, key, value)
+    if key == model.primary_key
+      {
+        "$ref" => "##{SCHEMAS_PATH}/ID"
+      }
+    elsif key.ends_with?("_id")
+      properties_value = {}
+      if GENERATOR_READ_ONLY_DEFINITIONS.include?(klass_name)
+        # Everything under providers data is read only for now
+        properties_value["$ref"] = "##{SCHEMAS_PATH}/ID"
+      else
+        properties_value["$ref"] = openapi_contents.dig(*path_parts(SCHEMAS_PATH), klass_name, "properties", key, "$ref") || "##{SCHEMAS_PATH}/ID"
+      end
+      properties_value
+    else
+      properties_value = {
+        "type" => "string"
+      }
+
+      case value.sql_type_metadata.type
+      when :datetime
+        properties_value["format"] = "date-time"
+      when :integer
+        properties_value["type"] = "integer"
+      when :float
+        properties_value["type"] = "number"
+      when :boolean
+        properties_value["type"] = "boolean"
+      when :jsonb
+        ['type', 'items'].each do |property_key|
```

Comment:
hmm you're missing this line https://github.com/ManageIQ/topological_inventory-api/blob/b9c30161c95b8fef9d85a002674ab04b7d796c3c/lib/tasks/swagger_generate.rake#L177

it was probably added after you took the generator?

Reply:
👍  Good catch!!  Thanks @Ladas for 👀 at this.  Will push an update.